### PR TITLE
Investigating raw scRNAseq data

### PR DESCRIPTION
**Pull request by andrewram4287**

### CODE REVIEW

**Comment by andrewram4287** [comment]
@aarmey It looks like the consistency of cell types and abundance seems to be very different depending on the experimental condition. My plan is to look over the cell type assignment again and see if there is anything I change on my end, since the single cell assignments don't look correct at the moment.

**Comment by aarmey** [comment]
@andrewram4287 I hadn't looked at the code you used here.

You are plotting the histogram of the *abundance* of specific genes across cell types and conditions. Are you sure you have enough of each cell type to reliably make a histogram?

**Comment by andrewram4287** [comment]
> @andrewram4287 I hadn't looked at the code you used here.
> 
> You are plotting the histogram of the _abundance_ of specific genes across cell types and conditions. Are you sure you have enough of each cell type to reliably make a histogram?

My initial thought was that there should be a consistent value across cell types. Would it make more sense to just plot the average then?

**Comment by aarmey** [comment]
Do you have enough cells like that? e.g., NK cells might be ~5% of your PBMC population, so if you only have 100 cells in a condition that's only 5 cells on average.

I don't know the answer here, I'm wondering.

**Comment by andrewram4287** [comment]
@aarmey This looks much more consistent.

**Comment by aarmey** [comment]
This looks better. I'd normalize by the total cell number with each drug because I know that varies. Alprostadil seems to be doing something weird...

File: sccp/data/scRNA_CellCount.csv [hunk 1]
```diff
@@ -0,0 +1,47 @@
+Drug,0
+Acetylcysteine,903
```
Comment (aarmey) [inline sccp/data/scRNA_CellCount.csv:2]: Can't this be calculated from the data itself?

**Review by aarmey — CHANGES_REQUESTED** [review]

File: sccp/data/scRNA_CellCount.csv (hunk 1 above)
Comment (andrewram4287) [reply sccp/data/scRNA_CellCount.csv:2]: I couldn't figure out a way since the tensor is based on the largest number of cells for an experiment.

Comment (aarmey) [reply sccp/data/scRNA_CellCount.csv:2]: But the filler rows are all zeros, so you can find them that way?

Longer-term, I have a half-finished PR to remove the need for this padding...

Comment (andrewram4287) [reply sccp/data/scRNA_CellCount.csv]: Used a different method now!

File: sccp/imports/scRNA.py [hunk 2]
```diff
@@ -184,7 +184,10 @@ def ThompsonXA_SCGenes(saveXA=False, offset=1.0):
     
     ### I *believe* that padding with zeros does not affect PARAFAC2 results.
     ### We should check this though.
-    XA.values -= np.nanmean(XA.values, axis=(0,1), keepdims=True)       
+    cellCount = np.count_nonzero(~np.isnan(XA[:, :, 0].to_numpy()), axis=1)
+    XA.values /= np.reshape(cellCount, (-1, 1, 1))
```
Comment (aarmey) [inline sccp/imports/scRNA.py:188]: Did you mean to add this different normalization? What are you aiming to do here? It looks like per-condition scaling by the number of cells?

Comment (andrewram4287) [reply sccp/imports/scRNA.py:188]: Yes, you commented before that you wanted me to "... normalize by the total cell number with each drug...", is this not what you meant?

Comment (aarmey) [reply sccp/imports/scRNA.py:188]: I meant normalize your plots of the cell counts within each condition, not the data itself.

**Pull request closed without merging**